airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/xmin/XminStateIterator.java

```diff
@@ -4,35 +4,29 @@
 
 package io.airbyte.integrations.source.postgres.xmin;
 
-import com.google.common.collect.AbstractIterator;
-import io.airbyte.integrations.source.postgres.internal.models.XminStatus;
-import io.airbyte.protocol.models.AirbyteStreamNameNamespacePair;
+import io.airbyte.cdk.integrations.source.relationaldb.state.SourceStateIterator;
+import io.airbyte.cdk.integrations.source.relationaldb.state.StateEmitFrequency;
 import io.airbyte.protocol.models.v0.AirbyteMessage;
 import io.airbyte.protocol.models.v0.AirbyteStateMessage;
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteStream;
+import java.time.Duration;
 import java.util.Iterator;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class XminStateIterator extends AbstractIterator<AirbyteMessage> implements Iterator<AirbyteMessage> {
+public class XminStateIterator extends SourceStateIterator<AirbyteMessage> {
```

Comment:
Remind me again : Why can't we get rid of `XminStateIterator` like we did the `CtidStateIterator`? We should try to get rid of the individual state iterators if possible

Reply:
mostly because of the error handling - if there's exception, xminStateIterator will not throw but just end gracefully. (Is that still expected?)